server/ethereum/ethereum.test.js

```diff
@@ -29,18 +30,87 @@ afterAll(() => {
   web3.currentProvider.connection.close()
 })
 
-// NOTE - this isn't really testing anything yet, just showing how to fire an event and getting the testing framework working
-test('firing an event pins a file', async done => {
+test('watcher pins file from registerData function', async done => {
   const testKey = web3.utils.fromAscii('testKey')
-  const testVal = { dag: 'killinnnnnn it' }
-  const hash = await node.dag.put(testVal)
+  const hash = await node.dag.put(dag)
 
   registerWatcher(contract)
   contract.methods
     .registerData(testKey, hash.toBaseEncodedString())
     .send({ from: accounts[0] }, () => {
-      setTimeout(() => {
+      setTimeout(async () => {
+        const pins = await node.pin.ls()
+        const match = pins.find(item => {
+          return item.hash === hash.toBaseEncodedString()
+        })
+        expect(match).toBeDefined()
         done()
       }, 2000)
     })
 })
+
+test('handlePinHashEvent pins file of cid it was passed', async done => {
+  dag.testKey = 'handlePinHashEventTestVal'
+  const hash = await node.dag.put(dag)
+  const eventObj = {
+    returnValues: {
+      cid: hash.toBaseEncodedString()
+    }
+  }
+
+  const res = await handlePinHashEvent(null, eventObj)
+  expect(res[0].hash).toBe(hash.toBaseEncodedString())
+  done()
+})
+
+test('handlePinHashEvent throws error with empty params', async done => {
+  try {
```

Comment:
could rewrite to be cleaner a la `state.test.js`

```
await expect(handlePinHashEvent()).rejects.toThrow(
   < your error here >
  )
```

Reply:
Yerr will do

Reply:
Yo, I fiddled with this for a couple mins but couldn't get it to work. If you feel like jumping in and giving it a crack, feel free. Otherwise I'd vote just to push it up.